### PR TITLE
03-Make: Implement make cooking soup

### PR DESCRIPTION
**Pull request by ghostzero2000**
Implement make cooking soup

### CODE REVIEW

File: 03-Make/cooking_soup/Makefile [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+	mkdir -vp ./$(FOOD_MARKET)/
+	mkdir -vp ./$(KITCHEN)/$(TABLE)/$(SPOON)
+	mkdir -vp ./$(KITCHEN)/$(TABLE)/$(KNIFE)
+	mkdir -vp ./$(KITCHEN)/$(TABLE)/$(POT)
+	mkdir -vp ./$(KITCHEN)/$(PRODUCTS)
+	mkdir -vp ./$(KITCHEN)/$(SLICED_PRODUCTS)
+	mkdir -vp ./$(DINNING_ROOM)
+	touch ./$(KITCHEN)/water.txt
+else
+	touch ./$(CREATE_WITHOUT_MAKE)
```
Comment (yekovalyov) [inline 03-Make/cooking_soup/Makefile]: What is the reason for this touch in cases of writing to file in the followed line. May be running chmod after stdout redirection is enough?

File: 03-Make/cooking_soup/Makefile [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+	chmod a+x ./$(CREATE_WITHOUT_MAKE)
+	@echo "#!/bin/bash" >> $(CREATE_WITHOUT_MAKE)
+	@echo "mkdir -vp ./$(FOOD_MARKET)/" >> $(CREATE_WITHOUT_MAKE)
+	@echo "mkdir -vp ./$(KITCHEN)/$(TABLE)/$(SPOON)" >> $(CREATE_WITHOUT_MAKE)
+	@echo "mkdir -vp ./$(KITCHEN)/$(TABLE)/$(KNIFE)" >> $(CREATE_WITHOUT_MAKE)
+	@echo "mkdir -vp ./$(KITCHEN)/$(TABLE)/$(POT)" >> $(CREATE_WITHOUT_MAKE)
+	@echo "mkdir -vp ./$(KITCHEN)/$(PRODUCTS)" >> $(CREATE_WITHOUT_MAKE)
+	@echo "mkdir -vp ./$(KITCHEN)/$(SLICED_PRODUCTS)" >> $(CREATE_WITHOUT_MAKE)
+	@echo "mkdir -vp ./$(DINNING_ROOM)" >> $(CREATE_WITHOUT_MAKE)
+	@echo "touch ./$(KITCHEN)/water.txt" >> $(CREATE_WITHOUT_MAKE)
```
Comment (yekovalyov) [inline 03-Make/cooking_soup/Makefile]: Please consider using statement 
cat <<LABEL >outfile
... multi-line text
LABEL

**Comment by ghostzero2000** [comment]
Hi. I updated the branch.

File: 03-Make/cooking_soup/Makefile (hunk 2 above)
Comment (ghostzero2000) [reply 03-Make/cooking_soup/Makefile]: Statement:
cat <<EOT >> test_3.txt
line 1
line 2
EOT

Work in a shell script, but I can not normal run in the make file.

File: 03-Make/cooking_soup/Makefile [hunk 3]
```diff
@@ -26,15 +26,36 @@ POT = pot
...
+mkdir -vp ./$(KITCHEN)/$(PRODUCTS)
+mkdir -vp ./$(KITCHEN)/$(SLICED_PRODUCTS)
+mkdir -vp ./$(DINNING_ROOM)
+endef
+export CREATE_DIRECTORIES_VAL
+
+
+
+
+
```
Comment (yekovalyov) [inline 03-Make/cooking_soup/Makefile]: trailing empty lines

**Comment by ghostzero2000** [comment]
Hello. I updated the current branch

File: 03-Make/cooking_soup/Makefile [hunk 4]
```diff
@@ -26,16 +26,31 @@ POT = pot
...
+#!/bin/bash
+mkdir -vp ./$(FOOD_MARKET)/
+mkdir -vp ./$(KITCHEN)/$(TABLE)/$(SPOON)
+mkdir -vp ./$(KITCHEN)/$(TABLE)/$(KNIFE)
+mkdir -vp ./$(KITCHEN)/$(TABLE)/$(POT)
+mkdir -vp ./$(KITCHEN)/$(PRODUCTS)
+mkdir -vp ./$(KITCHEN)/$(SLICED_PRODUCTS)
+mkdir -vp ./$(DINNING_ROOM)
+endef
+export CREATE_DIRECTORIES_VAL
```
Comment (yekovalyov) [inline 03-Make/cooking_soup/Makefile:42]: Is this export line required?

**Pull request merged**